File: Assets/Scripts/BombHudController.js
#pragma strict

private var playerController : PlayerController;
private var baseText : String;
private var textMesh : TextMesh;

function Start () {
	textMesh = transform.GetComponent(TextMesh);
	baseText = textMesh.text;
}

function Update () {
	textMesh.text = baseText + " (" + playerBombCount() + ")";
}

function playerBombCount() {
	return findPlayerController().bombCount;
}

function findPlayerController() {
	if (playerController) {
		return playerController;
	} else {
		playerController = GameObject.Find("Player(Clone)").GetComponent(PlayerController);
		return playerController;
	}
}

File: Assets/Scripts/BombHudController.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class BombHudController : MonoBehaviour
{
    private PlayerController playerController;
    private string baseText;
    private TextMesh textMesh;

    void Start()
    {
        textMesh = transform.GetComponent<TextMesh>();
        baseText = textMesh.text;
    }

    void Update()
    {
        textMesh.text = baseText + " (" + PlayerBombCount() + ")";
    }

    int PlayerBombCount()
    {
        return FindPlayerController().bombCount;
    }

    PlayerController FindPlayerController()
    {
        if (playerController)
        {
            return playerController;
        }
        else
        {
            playerController = GameObject.Find("Player(Clone)").GetComponent<PlayerController>();
            return playerController;
        }
    }
}
